File: backend-java/src/main/java/com/heliosync/healthcare/model/Consent.java
package com.heliosync.healthcare.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "consents")
public class Consent {
    @Id
    private String id;
    private String patientId;
    @org.springframework.data.mongodb.core.mapping.DocumentReference
    private Patient patient;
    private String doctorId;
    @org.springframework.data.mongodb.core.mapping.DocumentReference
    private Doctor doctor;
    private String hospitalId;
    @org.springframework.data.mongodb.core.mapping.DocumentReference
    private Hospital hospital;
    private String appointmentId;
    @org.springframework.data.mongodb.core.mapping.DocumentReference
    private Appointment appointment;
    private String purpose;
    private String status;
    private Boolean canViewHistory;
    private Boolean canViewVitals;
    private Boolean canViewPrescriptions;
    private Boolean canViewReports;
    private java.time.LocalDateTime expiresAt;
    private java.time.LocalDateTime grantedAt;
    private java.time.LocalDateTime revokedAt;
    private java.time.LocalDateTime createdAt;
}

File: frontend/src/services/consent.ts
import api from './api';
import { ApiResponse, Consent } from '../types';

export const consentService = {
  getConsents: async (params?: { patientId?: string; doctorId?: string }) => {
    const response = await api.get<ApiResponse<Consent[]>>('/consents', { params });
    return response.data;
  },

  createConsent: async (data: {
    patientId?: string;
    doctorId: string;
    hospitalId: string;
    appointmentId?: string;
    purpose?: string;
    canViewHistory?: boolean;
    canViewVitals?: boolean;
    canViewPrescriptions?: boolean;
    canViewReports?: boolean;
    durationDays?: number;
  }) => {
    const response = await api.post<ApiResponse<Consent>>('/consents', data);
    return response.data;
  },

  grantConsent: async (
    id: string,
    scopes?: {
      canViewHistory?: boolean;
      canViewVitals?: boolean;
      canViewPrescriptions?: boolean;
      canViewReports?: boolean;
    },
    durationDays?: number
  ) => {
    const response = await api.put<ApiResponse<Consent>>(`/consents/${id}/grant`, {
      scopes,
      durationDays,
    });
    return response.data;
  },

  rejectConsent: async (id: string) => {
    const response = await api.put<ApiResponse<Consent>>(`/consents/${id}/reject`);
    return response.data;
  },

  revokeConsent: async (id: string) => {
    const response = await api.put<ApiResponse<Consent>>(`/consents/${id}/revoke`);
    return response.data;
  },
};
